### PR TITLE
Scaling builds vertically

### PR DESCRIPTION
**Pull request by lassewesth**
This PR introduces running integration tests in parallel for the causal clustering module.

Because causal clustering integration tests spend significant time waiting, it is possible to run a lot of them in parallel - currently 2x number of cores for a massive 5x speedup in running time on reasonable hardware.

The gist of it is the introduction of PortAuthority as a global shared source of free port numbers to inject into servers, and fixing tests to take advantage of this.

### CODE REVIEW

File: enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/backup/BackupCoreIT.java [hunk 1]
```diff
@@ -102,7 +102,12 @@ static String backupAddress( GraphDatabaseFacade db )
         InetSocketAddress inetSocketAddress = db.getDependencyResolver()
                 .resolveDependency( Config.class ).get( CausalClusteringSettings.transaction_advertised_address )
                 .socketAddress();
-        return inetSocketAddress.getHostName() + ":" + (inetSocketAddress.getPort() + 2000);
+
+        HostnamePort hostnamePort = db.getDependencyResolver()
+                .resolveDependency( Config.class )
+                .get( OnlineBackupSettings.online_backup_server );
+
+        return inetSocketAddress.getHostName() + ":" + hostnamePort.getPort();
```
Comment (spacecowboy) [inline enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/backup/BackupCoreIT.java:110]: Need to read the changes more carefully - but this isn't using the PortAuthority? Should it?

Comment (lassewesth) [reply enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/backup/BackupCoreIT.java:110]: It does indirectly by reading `OnlineBackupSettings.online_backup_server`

Comment (lassewesth) [reply enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/backup/BackupCoreIT.java:110]: -and then not doing any arithmetics on it

Comment (lassewesth) [reply enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/backup/BackupCoreIT.java:110]: The old way was to take the Raft address and add 2000 to it. Now the backup port is set explicitly by test setup, and read here

**Review by davidegrohmann — APPROVED** [review]

File: enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/backup/BackupCoreIT.java (hunk 1 above)
Comment (spacecowboy) [reply enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/backup/BackupCoreIT.java:110]: I see. Great

**Pull request merged**